### PR TITLE
Medical Database : Create database base on a diagram

### PR DESCRIPTION
**Pull request by Zohra-Neda**
**In this project we :** 

- Create Medical-Clinic database according to the given diagram
- Create patient_treatments join table
- Add the FK indexes

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    treatment_id INT REFERENCES treatments(id)
+   
+);
+
+-- create join table
+CREATE TABLE patient_treatments (
+    id SERIAL PRIMARY KEY,
+    patient_id INT REFERENCES patients(id),
+    treatment_id INT REFERENCES treatments(id)
+);
```
Comment (gemmen29) [inline schema_based_on_diagram.sql]: - [ ] Kindly update the join table to match the [diagram](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/images/clinic_diagram.png) as the many-to-many relation is between the `medical_histories` and `treatments`.

### Note:
- Please don't forget to update the indexes based on the new update.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+REFERENCES treatments(id);
+
+-- add FK index for treatments_histories.treatment_id
+ALTER TABLE treatments_histories
+ADD CONSTRAINT fk_treatments_histories_treatments
+FOREIGN KEY (treatment_id)
+REFERENCES treatments(id);
+
+-- add FK index for treatments_histories.medical_histories_id
+ALTER TABLE treatments_histories
```
Comment (V-Blaze) [inline schema_based_on_diagram.sql]: - You have done great implementing some of the requirements and suggestions from the previous reviewer, but there are a few changes you need to make before moving to the next milestone. :+1: :100: 

- [ ] You have created the foreign keys in each table based on their relationships, as the previous reviewer suggested, you need to create INDEXES for all the foreign keys in each table. 
 - - EG: For `treatmeants_id` in invoices_items, you add index
 ```sh
CREATE INDEX ON invoices_items (treatments_id); 
```
- - For `medical_history_id` in  invoices, you add index
```sh
CREATE INDEX ON invoices (medical_history_id);
```
I'll suggest you u[update your code and do this for all the Foreign keys. :smile:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @Zohra-Neda, @hassanShakur  

# Almost there 🟡 

1094a3e620ebea4cd6cf2d0aceaa8f52


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing some changes from the previous reviewers, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Clean and organized codebase ✔
-  Correct gitflow👋 
-  Readme looks professional ✔

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KDlamini — APPROVED** [review]
### Hi @Zohra-Neda, @hassanShakur 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes for creating a database based on a diagram for the Vet clinic database. 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!🎊 🥳 

## Highlights 🌞 

✔️  Professional pull request
✔️  Correct use of Gitflow 
✔️  All required SQL commands are in their respective files

# Status: Approved ✔️ 

giphyapproved
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**